### PR TITLE
Add GitHub workflow

### PR DESCRIPTION
**Pull request by tsmock**
This does the following:
* Add PR and push build workflows
* Add scheduled update
* Add automatic NPM releases (note: this needs someone with access to https://www.npmjs.com/package/tag2link (@simon04 only?) to add an API key to GitHub secrets (`NPM_TOKEN`). This does require making new GH releases.
* Converts `build.js` to typescript and avoids updating the version/data_updated fields if nothing changed (so nothing gets committed)

### CODE REVIEW

**Comment by simon04** [comment]
Hi @tsmock, I've invited you as member to https://www.npmjs.com/package/tag2link.

**Comment by tsmock** [comment]
Thanks.

I don't have the ability to add a repository secret here, but I can at least do manual releases. :)

**Comment by simon04** [comment]
Hi @tsmock, I've invited added you as admin to tag2link. Thanks for maintaining!

**Comment by tsmock** [comment]
No problem -- ideally the release process will become fully automatic. I'll probably start working towards that at some point.

It isn't like this repository has a needs a bunch of code changes, just a release every now and then.

File: build.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env node
+import * as child_process from "child_process";
+import * as fs from "fs";
+import { isDeepStrictEqual } from "util";
```
Comment (simon04) [inline build.ts:4]: Hi @tsmock, have you possibly forgotten to commit the isDeepStrictEqual from util?

Comment (tsmock) [reply build.ts:4]: No. That should be in NodeJS 9+.

Comment (simon04) [reply build.ts:4]: Hi, sorry for the noise, my bad. Wasn't aware of such helpful utility functions. And tried to run the script using https://bun.sh/ which does not include isDeepStrictEqual.

Comment (tsmock) [reply build.ts:4]: No worries. I don't mind answering questions (and TIL that bun.sh exists). Anyway, it looks like a good chunk of the `util` functions are [not supported](https://github.com/oven-sh/bun/discussions/852) by bun.sh.

**Pull request merged**